### PR TITLE
add only-ipv4-support

### PR DESCRIPTION
**Pull request by 3Xpl0it3r**
add http server only bind  ipv4 address support

when I used Promtail ,I found Promtail alway bind ipv6, but I want to only bind in ipv4
I debug Promtail,i found Promtail used this library to build an http server.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -25,7 +25,7 @@ require (
 	github.com/sercand/kuberesolver v2.1.0+incompatible
 	github.com/sirupsen/logrus v1.4.2
 	github.com/stretchr/testify v1.4.0
-	github.com/uber/jaeger-client-go v2.22.1+incompatible
+	github.com/uber/jaeger-client-go v2.28.0+incompatible
```
Comment (bboreham) [inline go.mod:28]: Is this necessary to do ipv4-only?  If yes, please explain; if no, please move to a different PR.

File: server/server.go [hunk 2]
```diff
@@ -158,8 +164,13 @@ func New(cfg Config) (*Server, error) {
 	}, []string{"protocol"})
 	prometheus.MustRegister(tcpConnections)
 
+	// add ipv4 support
+	if cfg.Network == "" {
+		cfg.Network = DefaultNetwork
+	}
```
Comment (bboreham) [inline server/server.go:170]: Can you add to `RegisterFlags()` with the default please, instead of this?

File: server/server.go [hunk 3]
```diff
@@ -44,6 +49,7 @@ type SignalHandler interface {
 // Config for a Server
 type Config struct {
 	MetricsNamespace  string `yaml:"-"`
+	Network           string `yaml:"network"`
```
Comment (bboreham) [inline server/server.go:52]: For consistency this would be `http_listen_network`

**Review by bboreham — CHANGES_REQUESTED** [review]
Thanks for the PR.  The basic idea looks OK, but I don't think you should check in the files in `server/certs` - they are intended to be generated on-the-fly for testing only.

Maybe they should be added to `.gitignore`.

File: go.mod (hunk 1 above)
Comment (3Xpl0it3r) [reply go.mod:28]: no is not necessary, it looks like go mod replace it

File: server/server.go (hunk 2 above)
Comment (3Xpl0it3r) [reply server/server.go:170]: yeah, sure

File: server/server.go (hunk 3 above)
Comment (3Xpl0it3r) [reply server/server.go:52]: ok, I fix it

**Comment by 3Xpl0it3r** [comment]
@bboreham hi ,I reopen a new pr  #217 , this pr I will close it
